doc/whatsnew/fragments/5701.breaking

```diff
@@ -0,0 +1,4 @@
+``pylint`` is now equivalent to ``pylint .`` and won't show the help by default anymore.
+The help is still available with ``pylint --help`` or ``pylint --long-help``.
```

Comment:
Let's not promise equivalence with `pylint .`. I'm showing a behavior difference. Before, cd to an directory you haven't installed with pip, invoke `pylint .`, you get a quick lint of the `__init__.py` only; now you get the whole package linted.